packages/nameguard-js/src/separateGraphemes.ts

```diff
@@ -0,0 +1,27 @@
+import { splitGraphemes } from "text-segmentation";
+
+/**
+ * Separating graphemes with the usage of the external library - text-segmentation (https://github.com/niklasvh/text-segmentation)
+ * in colaboration with special Hangul treating.
+ *
+ * @param name - normalized domain name to separate graphemes from
```

Comment:
Added comment. This requirement simplifies the proof that our Hangul modification is correct. We had an idea to use ens-normalize to do custom splitting without this text-segmentation dep which would require normalized names. Because this is an internal function we can set this input restriction now which allows us to modify the implementation later.